packages/contracts/contracts/L1/teleportr/TeleportrDeposit.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.10;
+
+contract TeleportrDeposit {
+    address private owner;
+    uint256 private maxDepositAmount;
+    uint256 private minDepositAmount;
+    uint256 private maxBalance;
+    bool private canReceiveDeposit;
+    uint256 public totalDeposits;
+
+    constructor(
+        uint256 _maxDepositAmount,
+        uint256 _minDepositAmount,
+        uint256 _maxBalance,
+        bool _canReceiveDeposit
+    ) {
+        owner = msg.sender;
+        maxDepositAmount = _maxDepositAmount;
+        minDepositAmount = _minDepositAmount;
+        maxBalance = _maxBalance;
+        canReceiveDeposit = _canReceiveDeposit;
+        totalDeposits = 0;
+        emit OwnerSet(address(0), msg.sender);
+        emit MaxDepositAmountSet(0, _maxDepositAmount);
+        emit MinDepositAmountSet(0, _minDepositAmount);
+        emit MaxBalanceSet(0, _maxBalance);
+        emit CanReceiveDepositSet(_canReceiveDeposit);
+    }
+
+    // Send the contract's balance to the owner
+    function withdrawBalance() public isOwner {
+        uint256 balance = address(this).balance;
+        payable(owner).transfer(balance);
+        emit BalanceWithdrawn(owner, balance);
+    }
+
+    function destroy() public isOwner {
```

Comment:
I think we should get rid of the ability to `selfdestruct` this contract for simplicity's sake

Reply:
https://twitter.com/VitalikButerin/status/1489769962252091393?s=20&t=gu1WiV6EqRFyNQqy2efpxg